packages/twenty-server/src/modules/workflow/workflow-builder/workflow-step/utils/__tests__/insert-step.spec.ts

```diff
@@ -0,0 +1,88 @@
+import { insertStep } from 'src/modules/workflow/workflow-builder/workflow-step/utils/insert-step';
+import {
+  WorkflowAction,
+  WorkflowActionType,
+} from 'src/modules/workflow/workflow-executor/workflow-actions/types/workflow-action.type';
+
+describe('insertStep', () => {
+  const createMockAction = (
+    id: string,
+    nextStepIds?: string[],
+  ): WorkflowAction => ({
+    id,
+    name: `Action ${id}`,
+    type: WorkflowActionType.CODE,
+    settings: {
+      input: {
+        serverlessFunctionId: 'test',
+        serverlessFunctionVersion: '1.0.0',
+        serverlessFunctionInput: {},
+      },
+      outputSchema: {},
+      errorHandlingOptions: {
+        retryOnFailure: { value: false },
+        continueOnFailure: { value: false },
+      },
+    },
+    valid: true,
+    nextStepIds,
+  });
+
+  it('should insert a step at the end of the array when no parent or next step is specified', () => {
+    const step1 = createMockAction('1');
+    const step2 = createMockAction('2');
+    const newStep = createMockAction('new');
+
+    const result = insertStep([step1, step2], newStep);
+
+    expect(result).toEqual([step1, step2, newStep]);
+  });
+
+  it('should update parent step nextStepIds when inserting a step between two steps', () => {
+    const step1 = createMockAction('1', ['2']);
+    const step2 = createMockAction('2');
+    const newStep = createMockAction('new');
+
+    const result = insertStep([step1, step2], newStep, '1', '2');
+
+    expect(result).toEqual([
+      { ...step1, nextStepIds: ['new'] },
+      step2,
+      { ...newStep, nextStepIds: ['2'] },
+    ]);
```

Comment:
If I understand this example properly, it means that the steps will be in the following order: step1 -> newStep -> step2.

We assume that the order in the array will never be updated. I'm wondering if it won't hurt debugging. What do you think?